File: Python/ByTier/C/1971_C_Clock_and_Strings.py
# Problem: CF 1971 C - Clock and Strings
# https://codeforces.com/contest/1971/problem/C

"""
Problem: Determine if two strings connecting points on a clock intersect.
Algorithm/Technique: Modular arithmetic and comparison of relative positions.
Time Complexity: O(1) per test case.
Space Complexity: O(1) for variables, O(t) for output storage where t is number of test cases.

The approach is to fix one point (a) as the origin and compute the relative positions of the other points.
If both other points lie on the same side of the first point (when moving clockwise), the strings do not intersect.
Otherwise, they do intersect.
"""

import sys

input = sys.stdin.readline


def solve():
    a, b, c, d = map(int, input().split())
    # Normalize positions relative to 'a'
    b = (b - a) % 12
    c = (c - a) % 12
    d = (d - a) % 12

    # If both c and d are on the same side of b (when going clockwise from a),
    # then the strings do not intersect
    if (c > b and d > b) or (c < b and d < b):
        return "NO"
    else:
        return "YES"


T = int(input())
out = [solve() for _ in range(T)]
print("\n".join(map(str, out)))


# https://github.com/VaHiX/CodeForces/

File: CPP/ByTier/C/1971_C_Clock_and_Strings.cpp
// Problem: CF 1971 C - Clock and Strings
// https://codeforces.com/contest/1971/problem/C

/*
C. Clock and Strings
Algorithm: 
  - For each test case, we are given four distinct integers a, b, c, d representing positions on a clock labeled 1 to 12.
  - Alice connects a and b with a red string; Bob connects c and d with a blue string.
  - We need to determine if these two strings intersect on the clock.
  
  Key insight:
    - On a circular clock, two chords (strings) intersect if and only if one endpoint of each chord lies in the interval defined by the other chord's endpoints, considering wraparound.
    - To simplify comparison, we ensure that a <= b by swapping if necessary.
    - Then check whether c lies within [a,b] or d lies within [a,b]. If exactly one of these conditions is true, the chords intersect.

Time Complexity: O(1) per test case
Space Complexity: O(1)

Techniques:
  - Sorting intervals using conditional swaps
  - Checking if point is inside a circular interval using bounded comparisons
*/

#include <cstdio>
int main() {
  long t;
  scanf("%ld", &t);
  while (t--) {
    long a, b, c, d;
    scanf("%ld %ld %ld %ld", &a, &b, &c, &d);
    if (a > b) {
      long x = a;
      a = b;
      b = x;
    }
    // Check if c is in interval [a,b] (inclusive)
    bool u = (a <= c && c <= b);
    // Check if d is in interval [a,b] (inclusive)
    bool v = (a <= d && d <= b);
    // If exactly one of u or v is true, then intervals intersect
    puts(u ^ v ? "YES" : "NO");
  }
}


// https://github.com/VaHiX/codeForces/